### PR TITLE
Add Pod/Node Labels metric

### LINKED ISSUE #108: kube_node_info should add node labels
We run several node pools and use labels to differentiate them (as well as taints). One node pool in particular is built to be a canary for future versions of our underlying OS. I'd like to be able to build an alert the fires if that node goes away (assumably because some change in the lower env causes it's kubelet to fail). 

Being able to to say sum(kube_node_info{mynodelabel="canary"}) == 0 would be awesome.

### PR DESCRIPTION
**Pull request by brancz**
Closes #108

@fabxc @mxinden

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kube-state-metrics/130)

### CODE REVIEW

File: pod_test.go [hunk 1]
```diff
@@ -468,29 +470,48 @@ func TestPodCollector(t *testing.T) {
...
+						Namespace: "ns1",
+						Labels: map[string]string{
+							"app": "example",
+						},
+					},
+					Spec: v1.PodSpec{},
+				},
+			},
+			want: metadata + `
+				kube_pod_labels{kube_label_app="example",namespace="ns1",pod="pod1"} 1
```
Comment (fabxc) [inline pod_test.go]: Not prefixing the label names is unfortunately going to cause problems down the road – but maybe we can cut it down to just be `label_`. The metric itself is already prefixed with `kube_`.

**Comment by brancz** [comment]
@fabxc adapted according to your comment and added an info and label metric for `Service`s.

**Comment by fabxc** [comment]
👍

File: pod.go [hunk 2]
```diff
@@ -179,6 +201,31 @@ func (pc *podCollector) Collect(ch chan<- prometheus.Metric) {
...
+	}
+	return labelKeys, labelValues
+}
+
+func sanitizeLabelName(s string) string {
+	return invalidLabelCharRE.ReplaceAllString(s, "_")
+}
+
+func podLabelsDesc(labelKeys []string) *prometheus.Desc {
+	return prometheus.NewDesc(
```
Comment (andyxning) [inline pod.go]: @brancz It seems that prometheus will use `fqName`(`descPodLabelsName`) and `help`(`descPodLabels`) to uniquely define an metric. Once we first register it, the labels can be modified on the fly. Am i right?

Comment (andyxning) [inline pod.go]: Any docs about this behavior?

Comment (brancz) [inline pod.go]: This is actually just a validation that the go client library performs that we turned off. Metrics/time-series are _only_ identified by their unique label-set. The description registering was a premature validation attempt that is too hard of a validation in cases like this, although typically a correctly validation. So tl;dr yes you're right.

**Pull request merged**